File: mcp-server/tests/fixtures/trigger-runner/heartbeat.py
import json, os, sys, urllib.request

env = json.loads(sys.stdin.read())
secret = os.environ.get("CLAWDEVBOX_MCP_SECRET", "")
req = urllib.request.Request(
    env["callback_url"],
    data=json.dumps({"prompt": "python tick", "context": {"run_id": env["run_id"]}}).encode(),
    headers={"Content-Type": "application/json", "Authorization": f"Bearer {secret}"},
    method="POST",
)
with urllib.request.urlopen(req) as r:
    if r.status != 200:
        sys.stderr.write(f"status {r.status}\n"); sys.exit(1)
sys.stdout.write(json.dumps({"state": {"python": True}}))

File: mcp-server/tests/fixtures/trigger-runner/heartbeat.js
let body = '';
process.stdin.on('data', (c) => { body += c.toString('utf8'); });
process.stdin.on('end', async () => {
  const env = JSON.parse(body);
  const secret = process.env.CLAWDEVBOX_FIRE_SECRET || '';
  const res = await fetch(env.spawn_url, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json', Authorization: `Bearer ${secret}` },
    body: JSON.stringify({ prompt: 'node tick', context: { run_id: env.run_id } }),
  });
  if (!res.ok) { process.stderr.write(`status ${res.status}\n`); process.exit(1); }
  process.stdout.write(JSON.stringify({ state: { node: true } }));
});
